### ISSUE TITLE
matplotlib.units.ConversionError using datetime

### ISSUE THREAD

**Issue by moerchel** [Nuitka/Nuitka#1775]
I'm getting a ConversionError when using datetime as x axes values in matplotlib.

The used environment:
```
python -m nuitka --version
1.0.6
Commercial: None
Python: 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Flavor: Unknown
Executable: C:\temp\test\.venv\Scripts\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
```
```
python -m pip freeze
cycler==0.11.0
fonttools==4.37.1
kiwisolver==1.4.4
matplotlib==3.5.3
Nuitka==1.0.6
numpy==1.23.2
ordered-set==4.1.0
packaging==21.3
Pillow==9.2.0
pyparsing==3.0.9
python-dateutil==2.8.2
six==1.16.0
```
The setup of the virtual environment:
```
python -m venv .venv
.\.venv\Scripts\Activate.ps1
python -m pip install --upgrade pip
pip install --upgrade nuitka
pip install --upgrade matplotlib numpy ordered-set
```
The Nuitka options:
```
python -m nuitka --follow-imports test.py
```
or
```
python -m nuitka --enable-plugin=numpy --enable-plugin=tk-inter --standalone test.py
```
The code sample (test.py):
```python
import matplotlib.pyplot as plt
import numpy as np
import datetime as dt

buffer_size = 256

# the following is working
# ---
# x = np.linspace(0, 2 * np.pi, buffer_size)
# y = np.sin(x)
# ---

# this does not work
# ---
dtn = dt.datetime.now() - dt.timedelta(minutes=1)
x = [dtn + dt.timedelta(minutes=(x/buffer_size)) for x in range(0, buffer_size)]
y = np.linspace(0, 100, buffer_size)
# ---

fig, ax = plt.subplots()
ax.plot(x, y)
plt.show()
```
The output of test.cmd:
```
Traceback (most recent call last):
  File "C:\temp\test\.venv\lib\site-packages\matplotlib\axis.py", line 1506, in convert_units
    ret = self.converter.convert(x, self.units, self)
  File "C:\temp\test\.venv\lib\site-packages\matplotlib\dates.py", line 1922, in convert
    return self._get_converter().convert(*args, **kwargs)
SystemError: NULL result without error from call

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\temp\test\test.py", line 15, in <module>
    ax.plot(x, y)
  File "C:\temp\test\.venv\lib\site-packages\matplotlib\axes\_axes.py", line 1635, in plot
    lines = [*self._get_lines(*args, data=data, **kwargs)]
  File "C:\temp\test\.venv\lib\site-packages\matplotlib\axes\_base.py", line 312, in __call__
    yield from self._plot_args(this, kwargs)
  File "C:\temp\test\.venv\lib\site-packages\matplotlib\axes\_base.py", line 493, in _plot_args
    self.axes.xaxis.update_units(x)
  File "C:\temp\test\.venv\lib\site-packages\matplotlib\axis.py", line 1454, in update_units
    self._update_axisinfo()
  File "C:\temp\test\.venv\lib\site-packages\matplotlib\axis.py", line 1490, in _update_axisinfo
    self.set_default_intervals()
  File "C:\temp\test\.venv\lib\site-packages\matplotlib\axis.py", line 2254, in set_default_intervals
    xmin, xmax = self.convert_units(info.default_limits)
  File "C:\temp\test\.venv\lib\site-packages\matplotlib\axis.py", line 1509, in convert_units
    f'units: {x!r}') from e
matplotlib.units.ConversionError: Failed to convert value(s) to axis units: (datetime.date(2000, 1, 1), datetime.date(2010, 1, 1))
```

Do I need some special options in order to compile/run that code correctly?
The code works fine when running in Python or compiling just the main program:
```
python -m nuitka test.py
```

**Comment by kayhayen** [comment]
Perfect report, it seems that something is afoul with these converters, and needs a change in Nuitka. I will look into it.

**Comment by kkimirai** [comment]
i have the same problem,'matplotlib.units.ConversionError: Failed to convert value(s) to axis units: (datetime.date(2000, 1, 1), datetime.date(2010, 1, 1))'.when i want to use ax2.bar(time_series,P[:, 0])

**Comment by NickVeld** [comment]
It is not really related to this repo but maybe someone like me found this issue with the `ConversionError` .

I have solved the problem by `matplotlib.pyplot.scatter(x=list(df[DATE_COLUMN].astype('datetime64[us]').values), y=...)`

Basically, there was a check for the `datetime64[us]` type. If you convert data to this type upfront you will avoid the bugged code part.

**Comment by KevinRodriguez777** [comment]
this works nowadays, however, only if  `--enable-plugin=tk-inter` is used, it seems like the part in the plugin that queries matplotlib for the backend is broken
```
Traceback (most recent call last):
  File "C:\Users\Divine\Desktop\PyDis\MAIN~1.DIS\main.py", line 20, in <module>
  File "C:\Users\Divine\Desktop\PyDis\MAIN~1.DIS\matplotlib\pyplot.py", line 1613, in subplots
  File "C:\Users\Divine\Desktop\PyDis\MAIN~1.DIS\matplotlib\pyplot.py", line 934, in figure
  File "C:\Users\Divine\Desktop\PyDis\MAIN~1.DIS\matplotlib\pyplot.py", line 464, in new_figure_manager
  File "C:\Users\Divine\Desktop\PyDis\MAIN~1.DIS\matplotlib\pyplot.py", line 441, in _warn_if_gui_out_of_main_thread
  File "C:\Users\Divine\Desktop\PyDis\MAIN~1.DIS\matplotlib\pyplot.py", line 280, in _get_backend_mod
  File "C:\Users\Divine\Desktop\PyDis\MAIN~1.DIS\matplotlib\pyplot.py", line 342, in switch_backend
  File "importlib.py", line 126, in import_module
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'matplotlib.backends.backend_tkagg'
```

**Comment by kayhayen** [comment]
And without, I got this, very strange: `matplotlib.backends.backend_qtagg`, so I guess, it's the real issue, that there is no dependency on the backend. Not sure why we got away without the previously, but adding that it prompts me for using the Qt bindings, and I assume your tk-inter would be prompted and work. For me I would have to manually force tk-inter to be used, since with the backend selection as it is now, it will just use Qt and not work when I enable tk-inter, but it's sort of by design.
